client/src/App.js

```diff
@@ -59,45 +61,50 @@ function App() {
   }, [appView])
 
   return (
-    <div className='relative h-screen w-screen overflow-hidden'>
-      <MapElement
-        className={`relative left-10 w-[calc(100%-2.5rem)] transition-all ${
-          appView === AppView.NEW_TRIP ? 'h-full' : 'h-1/2'
-        }`}
-      />
-      <div
-        className={`relative left-10 flex w-[calc(100%-2.5rem)] flex-col justify-center ${
-          appView === AppView.NEW_TRIP ? 'h-0' : 'h-1/2'
-        }`}
-      >
-        {detailsContent}
-      </div>
-      <div
-        className={`fixed left-0 top-0 z-10 flex h-full w-1/5 flex-row overflow-hidden transition-all
-                ${isSidebarOpen ? 'left-0' : 'left-[-20vw]'}`}
-      >
-        <SideBar />
-      </div>
-      <span
-        className={`fixed top-0 z-10 flex h-full items-center rounded-none transition-all ${
-          isSidebarOpen ? 'left-[20vw]' : 'left-0'
-        }`}
-      >
-        <Button
-          onClick={() => dispatch(toggleSidebar())}
-          className='h-full w-10 rounded-none bg-slate-300 text-6xl hover:bg-slate-300'
-        >
-          {isSidebarOpen ? '‹' : '›'}
-        </Button>
-      </span>
-      {isSidebarOpen && (
-        <div
-          className={`fixed left-0 top-0 z-[5] h-full w-full bg-slate-800 opacity-30`}
-          onClick={() => dispatch(closeSidebar())}
-        />
+    <div>
+      {user ? ( // If user is logged in, show the app
+        <div className='relative h-screen w-screen overflow-hidden'>
+          <MapElement
+            className={`relative left-10 w-[calc(100%-2.5rem)] transition-all ${
+              appView === AppView.NEW_TRIP ? 'h-full' : 'h-1/2'
+            }`}
+          />
+          <div
+            className={`relative left-10 flex w-[calc(100%-2.5rem)] flex-col justify-center ${
+              appView === AppView.NEW_TRIP ? 'h-0' : 'h-1/2'
+            }`}
+          >
+            {detailsContent}
+          </div>
+          <div
+            className={`fixed left-0 top-0 z-10 flex h-full w-1/5 flex-row overflow-hidden transition-all
+                       ${isSidebarOpen ? 'left-0' : 'left-[-20vw]'}`}
+          >
+            <SideBar />
+          </div>
+          <span
+            className={`fixed top-0 z-10 flex h-full items-center rounded-none transition-all ${
+              isSidebarOpen ? 'left-[20vw]' : 'left-0'
+            }`}
+          >
+            <Button
+              onClick={() => dispatch(toggleSidebar())}
+              className='h-full w-10 rounded-none bg-slate-300 text-6xl hover:bg-slate-300'
+            >
+              {isSidebarOpen ? '‹' : '›'}
+            </Button>
+          </span>
+          {isSidebarOpen && (
+            <div
+              className={`fixed left-0 top-0 z-[5] h-full w-full bg-slate-800 opacity-30`}
+              onClick={() => dispatch(closeSidebar())}
+            />
+          )}
+        </div>
```

Comment:
I like what you did below with `<LoginScreen />`! Is it possible to move the main app code into a separate file like that too?

Reply:
Yeah great suggestion! I'll add this